Contributors-Guide/GlusterFS-Release-process.md

```diff
@@ -1,28 +1,116 @@
 
-1.  Add the release-notes to the docs/release-notes/ directory in the
-    sources
-2.  after merging the release-notes, create a tag like v3.6.2
-3.  push the tag to git.gluster.org
-4.  create the tarball with the [release job in
-    Jenkins](http://build.gluster.org/job/release/)
+## GlusterFS releases
+GlusterFS has 2 types of major releases, Long Term Maintenance (LTM) and Short Term Maintenance (STM). Major releases happen once every 3 months.
+
+Major releases don't guarantee complete backwards compatability with the previous major release.
+
+Minor releases will have guaranteed backwards compatibilty with earlier minor releases of the same branch.
+
+- LTM and STM state alternates with every release.
+- LTM and STM releases differ in the period of updates provided.
+
+### Long Term Maintenance (LTM) releases
+An LTM release reaches EOL when a LTM+2 release is done, 1 year later. LTM releases may get monthly minor releases for bug-fixes.
+
+### Short Term Maintenance (STM) releases
+STM releases are only supported until the next major release. STM releases may get monthly updates for bug fixes.
+
+The discussions around release schedules and lifecycles happened in the following mail threads on the GlusterFS maintainers mailing list.
+
+- https://www.gluster.org/pipermail/maintainers/2016-June/000892.html
+- https://www.gluster.org/pipermail/maintainers/2016-August/001174.html
+
+
+## GlusterFS major release window
+Each GlusterFS major release has a 3 month release window, in which changes get merged. The 3 month release window is split into two phases.
+
+1. A Open phase, where all changes get merged
+1. A Stability phase, where only changes that stabilize the release get merged.
+
+The first 2 months of a release window will be the Open phase, and the last month will be the stability phase.
+
+The release engineer (or team doing the release) is responsible for messaging.
+
+#### Open phase
+Any changes are accepted during this phase. New features that are introduced in this phase, need to be capable of being selectively built. All changes in the master branch are automatically incuded in the next release.
+
+All changes will be accepted during the Open phase. The changes have a few requirements,
+
+- a change fixing a bug SHOULD have public test case
+- a change introducing a new feature MUST have a disable switch that can disable the feature during a build
+
+
+#### Stability phase
+This phase is used to stabilize any new features introduced in the open phase, or general bug fixes for already existing features. A new release-<version> branch is created, all changes need to be sent to the master branch before getting backported to the new release branch. [kshlm: I prefer that the release gets tagged and branched at the end of the stability phase. This reduces the need for backports.]
+
+No new features will be merged in this phase.
 
```

Comment:
In the stability phase, we should also revert changes that are intended for future versions. http://review.gluster.org/14716 is one example of that. Other details in `configure.ac` need adjusting, like disabling experimental features and the like.